indico/modules/categories/controllers/display.py

```diff
@@ -167,3 +174,104 @@ def _process(self):
+
+        managers = sorted(self.category.get_manager_list(), key=lambda x: (x.principal_type.name, x.name))
+
+        def threshold_format(dt):
+            return dt.strftime('%Y-%m')
+
+        params = {'event_count': len(events),
+                  'events_by_month': events_by_month,
+                  'format_event_date': self.format_event_date,
+                  'future_event_count': future_event_count,
+                  'future_threshold': threshold_format(future_threshold),
+                  'happening_now': self.happening_now,
+                  'is_recent': self.is_recent,
+                  'managers': managers,
+                  'past_event_count': past_event_count,
+                  'past_events_by_month': past_events_by_month,
+                  'past_threshold': threshold_format(past_threshold)}
+        params.update(get_base_ical_parameters(session.user, self.category, 'category',
+                                               '/export/categ/{0}.ics'.format(self.category.id)))
+
+        if self.category.is_root:
+            if info.HelperMaKaCInfo.getMaKaCInfoInstance().isNewsActive():
+                news = [{'title': x.getTitle(), 'creation_dt': x.getCreationDate()} for x
+                        in ModuleHolder().getById('news').getNewsItemsList()[:2]]
+            else:
+                news = []
+
+            legacy_upcoming_events = ModuleHolder().getById('upcoming_events').getUpcomingEventList()
+            upcoming_events = [{'status': status,
+                                'start_dt': start_or_end_dt if not status == 'ongoing' else None,
+                                'end_dt': start_or_end_dt if status == 'ongoing' else None,
+                                'title': title,
+                                'id': event_id}
+                               for status, start_or_end_dt, title, event_id in legacy_upcoming_events]
+
+            return WPCategory.render_template('display/root_category.html', self.category, news=news,
+                                              upcoming_events=upcoming_events, **params)
+
+        return WPCategory.render_template('display/category.html', self.category, **params)
+
+
+class RHEventList(RHDisplayCategoryBase):
+    """Return the HTML for the event list before/after a specific month"""
+
+    @staticmethod
+    def _parse_year_month(string):
+        try:
+            dt = datetime.strptime(string, '%Y-%m')
+        except (TypeError, ValueError):
+            return
+        return DisplayTZ().getDisplayTZ(as_timezone=True).localize(dt)
+
+    def _checkParams(self):
+        RHDisplayCategoryBase._checkParams(self)
+        before = self._parse_year_month(request.args.get('before'))
+        after = self._parse_year_month(request.args.get('after'))
+        if before is None and after is None:
+            raise BadRequest(_('"before" or "after" parameter must be specified'))
```

Comment:
No need to translate this - you don't get this error unless you tinker around with the URL